Add TLS server and client cert auth support

tiran opened the pull request:
Custodia server now provides a HTTPS server implementation with optional
client cert verification.

CA cert and client cert/key pair can be set with additional parameters
and config options in Custodia's client and forwarder.

Signed-off-by: Christian Heimes <cheimes@redhat.com>

simo5 commented on custodia.conf:
```diff
@@ -1,6 +1,11 @@
 server_version = "Secret/0.0.7"
 debug = True
+#server_url = https://0.0.0.0:10443
+ssl_certfile = tests/ca/custodia-server.pem
```
can we use tls_ as the prefix for this and following parameters ?

simo5 commented on custodia/httpd/server.py:
```diff
@@ -85,6 +86,42 @@ def server_bind(self):
+        keyfile = self.config.get('ssl_keyfile')
+        cafile = self.config.get('ssl_cafile')
+        capath = self.config.get('ssl_capath')
+        verifymode = self.config.get('ssl_verifymode', 'CERT_NONE')
```
Shouldn't we have a stricter default here ?
Why CERT_NONE ?

tiran replied:
That is the server context. CERT_NONE disables client certs. With CERT_REQUIRED the client has to provide a valid client certificate that is signed by a trusted CA.

The client side always requires the server to have valid cert.

simo5 replied:
Can we verifymode -> client_cert then ? Or similar naming ?
I want to avoid confusion for admins.

tiran replied:
Let's follow KISS: *tls_verify_client*

True: ssl.CERT_REQUIRED
False: ssl.CERT_NONE

ssl.CERT_OPTIONAL is not that useful anyway.

simo5 commented on custodia/httpd/server.py:
```diff
@@ -85,6 +86,42 @@ def server_bind(self):
 
 
+class ForkingTLSServer(ForkingHTTPServer):
+    def __init__(self, server_address, handler_class, config, sslcontext=None):
```
sslcontext -> tls_context please

tiran replied:
Python's stdlib uses *context* as argument name for a SSLContext object.. python-requests will soon use *sslcontext*. Yet another spelling is just confusing. I'm going for *context* now.

simo5 commented:
Looks good, some nitpicks on naming, once we can fix those I see no problem pulling this in.

tiran commented:
I have addressed your nitpicking and simplified the client cert verification option.

Please merge PR #17 first. I'm going to rebase this patch on the new config parser and document the new option in the sphinx docs.

simo5 commented:
Now that PR #17 has been pushed, please rebase on top (also please make sure to remove added whitespaces, git am complained here.
